traffic_portal/app/src/common/modules/table/servers/TableServersController.js

```diff
@@ -17,405 +17,633 @@
-                clearCDNServerUpdates(cdn.id);
-            }, function () {
-                // do nothing
-            });
-        }
-    };
-
-    $scope.refresh = function() {
-        $state.reload(); // reloads all the resolves for the view
-    };
-
-    $scope.toggleVisibility = function(colName) {
-        const col = serversTable.column(colName + ':name');
-        col.visible(!col.visible());
-        serversTable.rows().invalidate().draw();
-    };
-
-    $scope.ssh = serverUtils.ssh;
-
-    $scope.isOffline = serverUtils.isOffline;
-
-    $scope.offlineReason = serverUtils.offlineReason;
-
-    $scope.getRelativeTime = dateUtils.getRelativeTime;
-
-    $scope.navigateToPath = locationUtils.navigateToPath;
-
-    var init = function () {
-        getStatuses();
-    };
-    init();
-
-    angular.element(document).ready(function () {
-        serversTable = $('#serversTable').DataTable({
-            "lengthMenu": [[25, 50, 100, -1], [25, 50, 100, "All"]],
-            "iDisplayLength": 25,
-            "aaSorting": [],
-            "columns": $scope.columns,
-            "initComplete": function(settings, json) {
-                try {
-                    // need to create the show/hide column checkboxes and bind to the current visibility
-                    $scope.columns = JSON.parse(localStorage.getItem('DataTables_serversTable_/')).columns;
-                } catch (e) {
-                    console.error("Failure to retrieve required column info from localStorage (key=DataTables_serversTable_/):", e);
-                }
-            }
-        });
-    });
+var TableServersController = function(servers, $scope, $state, $uibModal, $window, dateUtils, locationUtils, serverUtils, cdnService, serverService, statusService, propertiesModel, messageModel, userModel, $document) {
+	/**** Table cell formatters/renderers ****/
 
+	// browserify can't handle classes...
+	function SSHCellRenderer() {}
+	SSHCellRenderer.prototype.init = function(params) {
+		this.eGui = document.createElement("A");
+		this.eGui.href = "ssh://" + userModel.user.username + "@" + params.value;
+		this.eGui.setAttribute("target", "_blank");
+		this.eGui.textContent = params.value;
+	};
+	SSHCellRenderer.prototype.getGui = function() {return this.eGui;};
+
+	function UpdateCellRenderer() {}
+	UpdateCellRenderer.prototype.init = function(params) {
+		this.eGui = document.createElement("I");
+		this.eGui.setAttribute("aria-hidden", "true");
+		this.eGui.setAttribute("title", String(params.value));
+		this.eGui.classList.add("fa", "fa-lg");
+		if (params.value) {
+			this.eGui.classList.add("fa-clock-o");
+		} else {
+			this.eGui.classList.add("fa-check");
+		}
+	}
+	UpdateCellRenderer.prototype.getGui = function() {return this.eGui;};
+
+	/**
+	 * Gets text with which to file a status tooltip.
+	 * @returns {string | undefined} The offline reason if the server is offline, otherwise nothing.
+	 */
+	function offlineReasonTooltip(params) {
```

Comment:
does this work? i can't get it to when i hover over status and tbh not even really sure it's needed since there's an offline reason column anyhow.

Reply:
It only shows up when the server is set to an "offline" status, e.g. OFFLINE. Works for me. I don't really think it needs to be there either, but the jQuery datatables did it, so I did too.

Reply:
yep, that makes sense